modules/social_features/social_post/src/Hooks/SocialPostFormHooks.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\social_post\Hooks;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\hux\Attribute\Alter;
+use Drupal\social_group\CurrentGroupService;
+use Drupal\social_post\Service\SocialPostHelperInterface;
+
+
+final class SocialPostFormHooks {
```

Comment:
I initially wanted to have 1 class per hook replacement for better readability and maintenance but it will break the OS rule about not extra naming what we are doing. Because if I have 1 class per hook replacement the className would have the same name as the method.
IE: replacement of `social_post_form_post_form_alter`
ClassName `SocialPostFormPostFormAlterHook`, method called `SocialPostFormPostFormAlterHook`.
I'm really up for suggestion if we group hooks together or not in a class? (my preference is not) And then how do we name the class and the method?

I decided to name this class `SocialPostFormHooks ` meaning it will contains all hooks related to `form` but I initially wanted to have 1 class per hook

Reply:
My first instinct would be to split it in different classes. There will likely be cases where we it makes more sense to group them, e.g. when multiple hooks execute the same underlying code.

And I would ignore the rule about extra naming. Maybe it looks a bit weird, but while reading it you know what it does 🤷

Reply:
If I look at what the hook does in this case then it's preparing the post form to be personalised to the user and to the context?

What about `Drupal\social_post\Hooks\PostFormPersonalisation` as fully qualified name? It indicates it's doing something to the Post form and it's personalising. "Social" is omitted because that's included in the namespace already which IDEs also include in their class search.